make/photon/prepare/utils/migration.py

```diff
@@ -0,0 +1,44 @@
+import yaml
+import click
+import importlib
+import os
+
+from collections import deque
+
+from migrations import MIGRATION_BASE_DIR
+
+def read_conf(path):
+    with open(path) as f:
+        try:
+            d = yaml.safe_load(f)
+        except Exception as e:
+            click.echo("parse config file err, make sure your harbor config version is above 1.8.0", e)
+            exit(-1)
+    return d
+
+def _to_module_path(ver):
+    return "migrations.version_{}".format(ver.replace(".","_"))
+
+def search(input_ver: str, target_ver: str) -> deque :
+    """
+    Search accept a input version and the target version.
+    Returns the module of migrations in the upgrade path
+    """
+    upgrade_path, visited = deque(), set()
+    while True:
+        module_path = _to_module_path(target_ver)
+        visited.add(target_ver)  # mark current version for loop finding
+        if os.path.isdir(os.path.join(MIGRATION_BASE_DIR, 'version_{}'.format(target_ver.replace(".","_")))):
+            module = importlib.import_module(module_path)
+            if module.revision == input_ver:    # migration path found
+                break
+            elif module.down_revision is None:  # migration path not found
+                raise Exception('no migration path found')
```

Comment:
Not a blocker, but it would be helpful if you print more context in the errors.

For example what's the current migration path, and what 's the input/target version.

Same for other errors thrown in the `search` func.